README.md

```diff
@@ -1,2 +1,34 @@
+
+
+## Installing a module
+
+Each module is a self contained component and includes the source code, a general description, and instructions for building the the module with Maven and/or Gradle. In order to use one of these modules, clone this repo, build the module locally, and upload the jar using the spring xd shell `module upload` command. Detailed instructions are included with each project. You will find a good overview of modules and module packaging [here](http://docs.spring.io/spring-xd/docs/current/reference/html/#modules).
+
+## Updating the Spring XD version
+
+Each module project provides a build script (Maven, Gradle, or both) used to package the module as a jar. The build scripts are configured for a specific Spring XD version. If you intend to use a module with a different Spring XD version, rebuild the module configured for the target version. See the section on [module dependency management](http://docs.spring.io/spring-xd/docs/current-SNAPSHOT/reference/html/#module-dependency-management) in the Spring XD reference for details.
+
+## Developing Modules
+
+Read the [Creating a Module](http://docs.spring.io/spring-xd/docs/current-SNAPSHOT/reference/html/#creating-a-module) section in the Spring XD reference guide for a good overview of developing modules. If you want to dive into code, have a look at the example modules in the [spring-xd-samples][Samples] repository as well as the modules here. 
+
+### Should I provide both Maven and Gradle builds?
+
+Note that many of the [Samples][] provide both Maven and Gradle builds, primarily for illustration purposes. Some of the contributions here have followed this patterb. If you feel strongly about offering users a choice, we will respect that, but it's not really necessary. You are free to choose your preferred build tool. If you do decide to provide both, just be aware that you and whoever reviews your project will need to verify both builds are identically configured, complete successfully, and the jar contents produced by each are identical.
+
+## Contributing to Spring XD
+
+If you would like to contribute to this repository, we encourage contributions through pull requests from [forks of this repository](http://help.github.com/forking/). If you want to contribute code this way, please familiarize yourself with the process outlined for contributing to Spring projects here: [Contributor Guidelines](https://github.com/SpringSource/spring-integration/wiki/Contributor-Guidelines).
```

Comment:
"we encourage contributions through pull requests"  can be reworded "we encourage contributions via pull requests"